metaseq/launcher/tombyard.py

```diff
@@ -0,0 +1,126 @@
+            for tombstone_name in dirstones["scancel"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scancel {job_id} will be called every {period_after_tombstone_detected}
+                          until the job is dead"""
+                    )
+                    tombstone_detected = True
+                    period = period_after_tombstone_detected
+            for tombstone_name in dirstones["requeuehold"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scontrol requeuehold {job_id} will be called once.
+                         remove the file {tombstone_name} within the next {period_before_tombstone_detected}
+                         for it not to trigger the same command again"""
+                    )
+                    _ = os.popen(
+                        f"""
+                        scontrol requeuehold {job_id}
+                         """
+                    ).read()
+            for tombstone_name in dirstones["requeuehold"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scontrol release {job_id} will be called once.
+                         remove the file {tombstone_name} within the next {period_before_tombstone_detected}
+                         for it not to trigger the same command again"""
+                    )
+                    _ = os.popen(
+                        f"""
+                        scontrol release {job_id}
+                         """
+                    ).read()
+        if tombstone_detected:
+            _ = os.popen(
+                f"""
+                    scancel {job_id}
+                    """
+            ).read()
+        time.sleep(period.total_seconds())
+
+
+def tombstones(job_id, period=datetime.timedelta(seconds=60), dirstones=None):
+    if dirstones is None:
+        dirstones = {"scancel": [], "requeuehold": [], "release": []}
+        azure_base = "/shared/home"
+        for userdir in os.listdir(azure_base):
+            dirstones["scancel"].append(
+                os.path.join(azure_base, userdir, f"scancel_{job_id}")
+            )
+            dirstones["requeuehold"].append(
+                os.path.join(azure_base, userdir, f"requeuehold_{job_id}")
+            )
+            dirstones["release"].append(
+                os.path.join(azure_base, userdir, f"release_{job_id}")
+            )
+
+    for directive in ["scancel", "requeuehold", "release"]:
+        if directive not in dirstones.keys():
+            dirstones[directive] = []
+
+    # start a process that monitors
+    ctx = mp.get_context("spawn")
```

Comment:
this reads like the launch node is where this monitoring happens, and I'm not sure that's what we want. I think you just ported this code so let's make a task to fix this

Reply:
Yes, I just moved the file to allow for a unified way of importing, so I did not look at the details. But I will fix the whitespace and create an separate issue for this.